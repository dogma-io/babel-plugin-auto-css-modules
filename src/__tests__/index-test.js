import {transformFile} from '@babel/core'
import {join} from 'path'

function t(fileName) {
  const filePath = join(__dirname, 'fixtures', fileName)
  const options = {
    plugins: [require('../../lib/')],
    presets: ['@babel/env', '@babel/react'],
  }

  return new Promise((resolve, reject) => {
    transformFile(filePath, options, (err, result) => {
      if (err) {
        reject(err)
      } else {
        resolve(result)
      }
    })
  })
}

describe('babel-plugin-auto-css-modules', () => {
  it('should insert CSS import when CSS module exists with same name', () => {
    return t('foo.js').then(result => {
      expect(result.code).toMatchSnapshot()
    })
  })

  it('should not insert CSS import when CSS module does not exist with same name', () => {
    return t('bar.js').then(result => {
      expect(result.code).toMatchSnapshot()
    })
  })
})
