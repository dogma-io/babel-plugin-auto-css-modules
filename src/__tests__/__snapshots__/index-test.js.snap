// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-auto-css-modules should insert CSS import when CSS module exists with same name 1`] = `
"\\"use strict\\";

require(\\"./foo.css\\");

global.foo = 'baz';"
`;

exports[`babel-plugin-auto-css-modules should not insert CSS import when CSS module does not exist with same name 1`] = `
"\\"use strict\\";

global.bar = 'spam';"
`;
